### DISCUSSION TITLE
Draggable horizontal list

### DISCUSSION THREAD

**Discussion by ritz078** [adobe/react-spectrum#4506] [Q&A] [answered]
I have created a horizontal list and was able to successfully use Right and Left arrow keys to navigate using keyboardDelegate. I am not able to do the same for the drop indicator. It only works with Up/Down arrow keys. Any idea how I can achieve that ? I am using https://react-spectrum.adobe.com/react-aria/useDroppableCollection.html#reordering as the base implementation

https://github.com/adobe/react-spectrum/assets/5389035/13f66e22-43de-43ae-b751-f83ea608586c

**Comment by LFDanLu** [comment, marked as answer]
Unfortunately, useDroppableCollection doesn't support ArrowLeft/Right at the moment, only ArrowUp/Down are covered here: https://github.com/adobe/react-spectrum/blob/98cad3f064c5302c04a1140d12a2cacc3ee921a2/packages/%40react-aria/dnd/src/useDroppableCollection.ts#L538-L555. You would either need to extend the onKeyDown handling there to work with ArrowLeft/Right by default, add a option to customize that keydown behavior at the user's discretion (provide a keydown and chain it?), or implement your own hook that does something similar to useDroppableCollection but only accepts ArrowLeft/Right instead I'd imagine

**Reply by ritz078** [reply]
Great. I am sure I can come up with something based on your suggestions. Thank you.

**Reply by devongovett** [reply]
#4669 should add the necessary support for dnd in horizontally oriented lists.
